File: app/src/main/java/com/zhai/mvpdemo/MyApplication.java
package com.zhai.mvpdemo;

import android.app.Application;

/**
 * Created by zhai on 16/6/20.
 */
public class MyApplication extends Application {

    private AppComponent appComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        appComponent = DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .build();
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }
}

File: app/src/main/java/com/zhai/mvpdemo/MyApplication.kt
package com.zhai.mvpdemo

import android.app.Application
import android.content.Context

/**
 * Created by zhai on 16/6/20.
 */
class MyApplication : Application() {
    // 延迟初始化
    lateinit var appComponent: AppComponent

    override fun onCreate() {
        super.onCreate()
        appComponent = DaggerAppComponent.builder().appModule(AppModule(this)).build()
    }

    companion object {
        @JvmStatic
        fun getComponent(context: Context): AppComponent {
            val app = context.applicationContext as MyApplication
            return app.appComponent
        }
    }
}

